src/zenml/pipelines/pipeline_definition.py

```diff
@@ -806,31 +824,83 @@ def _run(
 
         with track_handler(AnalyticsEvent.RUN_PIPELINE) as analytics_handler:
             stack = Client().active_stack
-            deployment = self._create_deployment(**self._run_args)
-
-            self.log_pipeline_deployment_metadata(deployment)
-            run = create_placeholder_run(deployment=deployment)
 
-            analytics_handler.metadata = self._get_pipeline_analytics_metadata(
-                deployment=deployment,
-                stack=stack,
-                run_id=run.id if run else None,
+            compiled_deployment, schedule, build = self._compile(
+                **self._run_args
             )
+            self._run_args["deployment"] = compiled_deployment
+            self._run_args["schedule"] = schedule
+            self._run_args["build"] = build
```

Comment:
This is super ugly 😅 I'm guessing this was necessary to figure out the final state of `enable_pipeline_logs` before calling the rest of the code?

I'm wondering if there's a cleaner way to achieve this? I guess the value is just:
```
logging_enabled = run_config.enable_pipeline_logs
if logging_enabled is None:
  logging_enabled = self.config.enable_pipeline_logs
```

Reply:
Not just super-ugly, but apparently also breaking everything.

Your suggestion looks a lot better. Done.